### PR TITLE
release/dist/synology: build synology packages with cmd/dist

### PR DESCRIPTION
**Pull request by danderson**
Updates #8217

---

Because of cmd/dist's memoizing and build cache sharing, the build is faster. Before:

```
$ time make spkall
...
make spkall  34.45s user 7.29s system 229% cpu 18.183 total
```

After:

```
$ go run ./cmd/dist build synology
...
go run ./cmd/dist build synology  27.50s user 5.43s system 957% cpu 3.437 total
```

I compared the outputs of the old and new implementation. They are bit for bit identical, except that cmd/dist built binaries are a bit larger due to extra version info (a dozen bytes or so), and cmd/dist compresses the packages slightly better (~1k smaller to distribute).

### CODE REVIEW

File: release/dist/synology/files/config [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+	".url": {
+		"SYNO.SDS.Tailscale": {
+			"type": "url",
+			"version": "1.8.3",
```
Comment (sailorfrag) [inline release/dist/synology/files/config:5]: Does this need to be generated with the current version number?

File: release/dist/synology/files/Tailscale.sc
Comment (sailorfrag) [inline release/dist/synology/files/Tailscale.sc:1]: Is missing trailing EOL required for this file?

There are several other files showing this too. I recognize that these are identical to what was in the `tailscale/tailscale-synology` repo and thus it's not a new concern.

File: cmd/dist/dist.go [hunk 2]
```diff
@@ -13,15 +13,28 @@ import (
...
 )
 
+var synologyPackageCenter bool
+
 func getTargets() ([]dist.Target, error) {
-	return unixpkgs.Targets(), nil
+	var ret []dist.Target
+
+	ret = append(ret, unixpkgs.Targets()...)
+	ret = append(ret, synology.Targets(synologyPackageCenter)...)
```
Comment (sailorfrag) [inline cmd/dist/dist.go:26]: Is this supposed to be conditional on `synologyPackageCenter`?

File: release/dist/synology/pkgs.go [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+	f := func(k, v string) {
+		fmt.Fprintf(&ret, "%s=%q\n", k, v)
+	}
+	f("package", "Tailscale")
+	f("version", fmt.Sprintf("%s-%d", b.Version.Short, b.Version.Synology[t.dsmMajorVersion]))
+	f("arch", t.filenameArch)
+	f("description", "Connect all your devices using WireGuard, without the hassle.")
+	f("displayname", "Tailscale")
+	f("maintainer", "Tailscale, Inc.")
+	f("maintainer_url", "https://github.com/tailscale/tailscale-synology")
```
Comment (sailorfrag) [inline release/dist/synology/pkgs.go:105]: Is this still correct?

**Review by sailorfrag — APPROVED** [review]
Seems mostly fine, likely some trivial changes needed

File: release/dist/synology/files/config (hunk 1 above)
Comment (danderson) [reply release/dist/synology/files/config:5]: It's a TODO in the code I ported over from tailscale-synology. I decided to not get to the bottom of it in this change, so I left it as-is and ported the TODO to figure it out.

File: release/dist/synology/files/Tailscale.sc
Comment (danderson) [reply release/dist/synology/files/Tailscale.sc:1]: I have no idea offhand. I can't imagine that synology software would care... But I'm also afraid to find out now :)

File: cmd/dist/dist.go (hunk 2 above)
Comment (danderson) [reply cmd/dist/dist.go:26]: No, the bool is specifying which variant of the packages to build: standalone for sideloading, or the ones for distribution in the synology package center. The latter can request permission to use tuntap devices and such, whereas the sideloaded ones can't. Effectively, we're the only consumers of this flag, and we'll set it when generating packages for Synology to distribute, as opposed to the ones we put on pkgs.tailscale.com so people can sideload.

Comment (danderson) [reply cmd/dist/dist.go:36]: Added a comment to expand on this at the call site.

File: release/dist/synology/pkgs.go (hunk 3 above)
Comment (danderson) [reply release/dist/synology/pkgs.go]: Woops, no it's not. Fixed.

**Pull request merged**